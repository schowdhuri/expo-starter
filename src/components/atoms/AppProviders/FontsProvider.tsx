import {
  AlegreyaSansSC_100Thin,
  AlegreyaSansSC_100Thin_Italic,
  AlegreyaSansSC_300Light,
  AlegreyaSansSC_300Light_Italic,
  AlegreyaSansSC_400Regular,
  AlegreyaSansSC_400Regular_Italic,
  AlegreyaSansSC_500Medium,
  AlegreyaSansSC_500Medium_Italic,
  AlegreyaSansSC_700Bold,
  AlegreyaSansSC_700Bold_Italic,
  AlegreyaSansSC_800ExtraBold,
  AlegreyaSansSC_800ExtraBold_Italic,
  AlegreyaSansSC_900Black,
  AlegreyaSansSC_900Black_Italic,
} from "@expo-google-fonts/alegreya-sans-sc";
import {
  Roboto_100Thin,
  Roboto_100Thin_Italic,
  Roboto_300Light,
  Roboto_300Light_Italic,
  Roboto_400Regular,
  Roboto_400Regular_Italic,
  Roboto_500Medium,
  Roboto_500Medium_Italic,
  Roboto_700Bold,
  Roboto_700Bold_Italic,
  Roboto_900Black,
  Roboto_900Black_Italic,
} from "@expo-google-fonts/roboto";
import { useFonts } from "expo-font";
import type { FC } from "react";

export const FontsProvider: FC = ({ children }) => {
  const [fontsLoaded] = useFonts({
    AlegreyaSansSC_100Thin,
    AlegreyaSansSC_100Thin_Italic,
    AlegreyaSansSC_300Light,
    AlegreyaSansSC_300Light_Italic,
    AlegreyaSansSC_400Regular,
    AlegreyaSansSC_400Regular_Italic,
    AlegreyaSansSC_500Medium,
    AlegreyaSansSC_500Medium_Italic,
    AlegreyaSansSC_700Bold,
    AlegreyaSansSC_700Bold_Italic,
    AlegreyaSansSC_800ExtraBold,
    AlegreyaSansSC_800ExtraBold_Italic,
    AlegreyaSansSC_900Black,
    AlegreyaSansSC_900Black_Italic,
    Roboto_100Thin,
    Roboto_100Thin_Italic,
    Roboto_300Light,
    Roboto_300Light_Italic,
    Roboto_400Regular,
    Roboto_400Regular_Italic,
    Roboto_500Medium,
    Roboto_500Medium_Italic,
    Roboto_700Bold,
    Roboto_700Bold_Italic,
    Roboto_900Black,
    Roboto_900Black_Italic,
  });
  if (!fontsLoaded) {
    return <></>;
  }
  return <>{children}</>;
};
