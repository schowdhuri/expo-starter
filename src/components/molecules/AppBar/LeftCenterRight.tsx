import { HStack } from "native-base";

interface LeftCenterRightProps {
  CenterComponent?: JSX.Element;
  LeftComponent?: JSX.Element;
  RightComponent?: JSX.Element;
}

export function LeftCenterRight(props: LeftCenterRightProps) {
  const {
    CenterComponent,
    LeftComponent,
    RightComponent: RightComonent,
  } = props;
  return (
    <HStack alignItems="center" w="100%">
      {LeftComponent}
      {CenterComponent}
      {RightComonent}
    </HStack>
  );
}
