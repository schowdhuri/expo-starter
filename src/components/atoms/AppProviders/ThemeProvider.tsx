import { NativeBaseProvider } from "native-base";
import type { FC } from "react";
import { theme } from "@constants/theme";
import { FontsProvider } from "./FontsProvider";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <FontsProvider>
      <NativeBaseProvider initialWindowMetrics={INSET} theme={theme}>
        {children}
      </NativeBaseProvider>
    </FontsProvider>
  );
};

const INSET = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};
