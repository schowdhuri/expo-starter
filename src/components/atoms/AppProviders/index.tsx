import { NavigationContainer } from "@react-navigation/native";
import type { FC } from "react";
import { QueryClient, QueryClientProvider } from "react-query";
import { RecoilRoot } from "recoil";
import { ThemeProvider } from "./ThemeProvider";

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: 30000,
      refetchOnWindowFocus: false,
    },
  },
});

export const AppProviders: FC = ({ children }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <RecoilRoot>
        <ThemeProvider>
          <NavigationContainer>{children}</NavigationContainer>
        </ThemeProvider>
      </RecoilRoot>
    </QueryClientProvider>
  );
};
