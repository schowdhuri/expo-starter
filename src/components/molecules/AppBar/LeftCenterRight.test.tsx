import { render, waitFor } from "@testing-library/react-native";
import { Box, Text } from "native-base";
import { AppProviders } from "@atoms/AppProviders";
import { LeftCenterRight } from "./LeftCenterRight";

describe("molecules::LeftCenterRight", () => {
  test("renders without exploding", async () => {
    const { toJSON, getByText, unmount } = render(
      <AppProviders>
        <LeftCenterRight
          LeftComponent={
            <Box flex={2}>
              <Text>LeftComponent</Text>
            </Box>
          }
          CenterComponent={
            <Box flex={3}>
              <Text>CenterComponent</Text>
            </Box>
          }
          RightComponent={
            <Box flex={2} alignItems="flex-end">
              <Text>RightComponent</Text>
            </Box>
          }
        />
      </AppProviders>
    );
    await waitFor(() => expect(getByText("LeftComponent")).toBeDefined());
    expect(toJSON()).toMatchSnapshot();
    unmount();
  });
});
