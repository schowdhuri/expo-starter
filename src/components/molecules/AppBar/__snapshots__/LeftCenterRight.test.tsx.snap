// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules::LeftCenterRight renders without exploding 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": "AlegreyaSansSC_400Regular",
            "fontSize": 14,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        LeftComponent
      </Text>
    </View>
    <View
      dataSet={Object {}}
      style={Object {}}
    />
    <View
      dataSet={Object {}}
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": "AlegreyaSansSC_400Regular",
            "fontSize": 14,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        CenterComponent
      </Text>
    </View>
    <View
      dataSet={Object {}}
      style={Object {}}
    />
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 2,
        }
      }
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": "AlegreyaSansSC_400Regular",
            "fontSize": 14,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        RightComponent
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
